### PR TITLE
[Docs] Convert QMK Docs over to mkdocs

### PR DESCRIPTION
**Pull request by drashna**
## Types of Changes

- [x] Core
- [x] New feature
- [x] Enhancement/optimization
- [x] Documentation

## Issues Fixed or Closed by This PR

* beta.docs.qmk.fm

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by peepeetee** [comment]
What tooling was used to convert the .md files?
We have about 20 docs translation PRs in flight, I presume we need to update them

**Comment by drashna** [comment]
Mostly, it was moving the files around, and then changing `!>` and `?>` to `!!! (keyword)`. 

for tags, it's `:id=(tag)` to `{: id=(tag) }`.

So, a simple string replace could be used to update PRs.  And rebasing them may be all that is actually needed (aside from the string replace)

**Comment by peepeetee** [comment]
Also, I'm here to raise the concern for zh, as we have #15444 in flight that attempts to add zh-TW (right now it's just README.md and newbs.md). So we need a zh-CN and a zh-TW. Could that be accommodated?

**Comment by drashna** [comment]
That shouldn't be an issue.  It looks like it supports zh (simplified chinese), and zh-tw (chinese, tiawanese).

https://squidfunk.github.io/mkdocs-material/setup/changing-the-language/#site-language

**Comment by peepeetee** [comment]
I would prefer simplified Chinese to not be called zh, but I guess that a PR with them not us

**Comment by peepeetee** [comment]
简体中文 is the simplified chinese endonym

**Review by tzarc — CHANGES_REQUESTED** [review]
Should the scripts be put into `util/` in general to reduce clutter?
Also, can't comment on moves with no content changes -- what about the `.nojekyll` and `CNAME` files? Are they still required?

File: site/index.html [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<!DOCTYPE html>
+<html>
+<head>
+  <meta charset="utf-8">
+  <title>Redirecting</title>
+  <noscript>
+    <meta http-equiv="refresh" content="1; url=latest/" />
```
[the remark this answers is not in the document]
Comment (tzarc) [reply site/index.html:7]: Perhaps `master/en`?

File: build_translations.sh [hunk 2]
```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env bash
```
[the remark this answers is not in the document]
Comment (drashna) [reply build_translations.sh:1]: Yeah, should probably be moved.  will see about doing so, and testing that this week.

[the remark this answers is not in the document]
Comment (drashna) [reply build_translations.sh]: Addressed and moved.

File: summary_to_nav [hunk 3]
```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env python3
```
[the remark this answers is not in the document]
Comment (drashna) [reply summary_to_nav]: Nope, removed.

**Comment by drashna** [comment]
Note:  This will require an update to the gh-pages repo, in the form of adding an `index.html` file that redirects to `master/en`

File: site/index.html (hunk 1 above)
[the remark this answers is not in the document]
Comment (drashna) [reply site/index.html]: Yup, that works

**Pull request closed without merging**